pydantic_settings/sources.py

```diff
@@ -307,8 +317,24 @@ def explode_env_vars(
             env_name_without_prefix = env_name[self.env_prefix_len :]
             _, *keys, last_key = env_name_without_prefix.split(self.env_nested_delimiter)
             env_var = result
+            target_field: Optional[FieldInfo] = field
+
             for key in keys:
+                target_field = self.next_field(target_field, key)
                 env_var = env_var.setdefault(key, {})
+
+            # get proper field with last_key
+            target_field = self.next_field(target_field, last_key)
+
+            # check if env_val maps to a complex field and if so, parse the env_val
+            if target_field and env_val:
+                is_complex, allow_json_failure = self._field_is_complex(target_field)
+                if is_complex:
```

Comment:
do we have a way to disable json parsing?

Reply:
I've tried to disable it here but it doesn't work
I haven't looked deeply on it but seems we need it here